src/software/ai/hl/stp/play/enemy_ball_placement/enemy_ball_placement_play_fsm.cpp

```diff
@@ -0,0 +1,139 @@
+          std::make_shared<MoveTactic>(),
+          std::make_shared<MoveTactic>(),
+      }),
+      goalie_tactic(std::make_shared<GoalieTactic>(ai_config)),
+      distance_to_keep(ENEMY_BALL_PLACEMENT_DISTANCE_METERS +
+                       4 * ROBOT_MAX_RADIUS_METERS),
+      nearly_placed_threshold(0.5)
+{
+}
+
+bool EnemyBallPlacementPlayFSM::hasPlacementPoint(const Update& event)
+{
+    return event.common.world_ptr->gameState().getBallPlacementPoint().has_value();
+}
+
+void EnemyBallPlacementPlayFSM::setPlacementPoint(const Update& event)
+{
+    placement_point = event.common.world_ptr->gameState().getBallPlacementPoint().value();
+}
+
+bool EnemyBallPlacementPlayFSM::isNearlyPlaced(const Update& event)
+{
+    return (event.common.world_ptr->ball().position() - placement_point).length() <
+           nearly_placed_threshold;
+}
+
+void EnemyBallPlacementPlayFSM::avoid(const Update& event)
+{
+    WorldPtr world_ptr                  = event.common.world_ptr;
+    PriorityTacticVector tactics_to_run = {{}};
+    Point ball_pos                      = world_ptr->ball().position();
+
+    Stadium stadium                = Stadium(ball_pos, placement_point, distance_to_keep);
+    Vector placement_point_to_ball = ball_pos - placement_point;
+
+    // Check if robots are inside the ball placement stadium
+    int idx = 0;
+    for (auto const& robot : world_ptr->friendlyTeam().getAllRobots())
+    {
+        if (contains(stadium, robot.position()))
+        {
+            Vector lateral_positioning_vector =
+                placement_point_to_ball.perpendicular().normalize();
+            Vector longitudinal_positioning_vector =
+                (robot.position() - placement_point).project(placement_point_to_ball);
+            Point p1 = placement_point + longitudinal_positioning_vector +
+                       lateral_positioning_vector * distance_to_keep;
+            Point p2 = placement_point + longitudinal_positioning_vector -
+                       lateral_positioning_vector * distance_to_keep;
+            Point destination;
+            Rectangle fieldLines = world_ptr->field().fieldLines();
+            if (!contains(fieldLines, p1))
+            {
+                destination = p2;
+            }
+            else if (!contains(fieldLines, p2))
+            {
+                destination = p1;
+            }
+            else
+            {
+                destination =
+                    distance(p1, robot.position()) < distance(p2, robot.position()) ? p1
+                                                                                    : p2;
+            }
+            avoid_interference_tactics[idx]->updateControlParams(
+                destination, (ball_pos - robot.position()).orientation(), 0);
+        }
+        else
+        {
+            avoid_interference_tactics[idx]->updateControlParams(
+                robot.position(), (ball_pos - robot.position()).orientation(), 0);
+        }
```

Comment:
Could you add some comments clarifying what youre calculating and why youre deciding on which point to pick as the destination?